refactor(chat): show inline code more subtle

elianiva opened the pull request:
## Context

I've been wanting to change this for a while. The current inline code feels a bit 'out of place', every time the LLM outputs them I find it hard to read and just clutters the chat.

## Implementation

- Adjusted the monospace font to be _slightly_ smaller than the rest because they tend to appear larger just a tiny bit. 
- Adjusted the border colour so they are more subtle, the old one makes it as if these blocks are clickable and a separate element instead of just plain text with different styling

## Screenshots

Epilepsy warning, I'm switching between multiple different themes (dark and light) to test it. 

### Before
https://github.com/user-attachments/assets/2d57bce0-294a-49d7-a749-09e4e0317909

### After
https://github.com/user-attachments/assets/65a0e785-9c48-4092-93e9-906f53e9d002

## How to Test

Try having a chat with the LLM and ask it to output some inline code

## Get in Touch

@elianiva on Roocode discord server

<!-- ELLIPSIS_HIDDEN -->

----

> [!IMPORTANT]
> Refactor inline code styling in `MarkdownBlock.tsx` to be more subtle by adjusting font size and border color.
> 
>   - **Styling Changes**:
>     - In `MarkdownBlock.tsx`, adjusted `code:not(pre > code)` to have `font-size: 0.9em` for a slightly smaller appearance.
>     - Changed `border` color of `code:not(pre > code)` to `var(--vscode-tab-border, #424242)` for a more subtle look.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=RooVetGit%2FRoo-Code&utm_source=github&utm_medium=referral)<sup> for 7791b8eb6fc5146b954fe789fc5510398c9a7920. It will automatically update as commits are pushed.</sup>

<!-- ELLIPSIS_HIDDEN -->

cte commented:
This bugs me too! Let's figure out the font size thing though.

adamhill commented:
I like the font change!

 One thing that helps me see the code (inline) with normal text is a much more vibrant / saturated color, especially in darker modes. (I have been trained by GH Copilot choices)

Are the chat blocks picking up the colors from the theme the user has selected? And would actual large code blocks pick up the theme proper from whatever component we use?

elianiva commented:
> I like the font change!
> 
> One thing that helps me see the code (inline) with normal text is a much more vibrant / saturated color, especially in darker modes. (I have been trained by GH Copilot choices)
> 
> Are the chat blocks picking up the colors from the theme the user has selected? And would actual large code blocks pick up the theme proper from whatever component we use?

i guess we need to figure out which token that consistently has bright/stand out colour across colourschemes

as for the codeblock, there is some work at https://github.com/RooVetGit/Roo-Code/pull/1328
i believe it will take the current theme since it uses shiki (which is using the same textmate grammar that vscode uses, so it will be compatible)

elianiva commented:
i played with this a bit more, a few changes:
- reverted the font size change, apparently the bigger font size only happens on low resolution? not sure, but it doesn't happen now after I changed my monitor
- removed the border since it makes the inline code looks like its own entity, we can utilise borders for actually clickable items
- adjusted the foreground and background colour, now it's similar with Github copilot which IMO is better because it handles multiple themes better especially with the high contrast ones

I think this PR is ready for review.

updated preview:

### Before
https://github.com/user-attachments/assets/cf0b883c-b492-4c64-912b-2c1c9665df81

### After
https://github.com/user-attachments/assets/5e98477f-3517-409e-b579-af8cabb511a7

cte commented:
Thanks!
